src/snakeGame/SnakeGame.java

```diff
@@ -0,0 +1,191 @@
+package snakeGame;
+
+import java.awt.Color;
+
+import edu.macalester.graphics.CanvasWindow;
+import edu.macalester.graphics.FontStyle;
+import edu.macalester.graphics.GraphicsGroup;
+import edu.macalester.graphics.GraphicsText;
+
+// Authors:           Precious Dlamini, Ella Morrow, and Abbie Natkin
+// Class:             COMP 127 - 01
+// Acknowledgements:  Thanks to Paul Cantrell for helping us work out a few bugs.
+
+/** This is the main class that will set up and run a game of "Snake". */
+
+public class SnakeGame {
+    private static final int CANVAS_WIDTH = 600;
+    private static final int CANVAS_HEIGHT = 600;
+    private CanvasWindow canvas;
+    private Snake snake;
+    private GraphicsGroup gameLayer;
+    private Boolean animating;
+    private MushroomManager mushroomManager;
+    private Player player;
+    private int levels = 1;
+    private GraphicsText score;
+    private GraphicsText lives;
+    private GraphicsText level;
+
+    public SnakeGame() {
+        canvas = new CanvasWindow("Snake Game!", CANVAS_WIDTH, CANVAS_HEIGHT);
+        canvas.setBackground(new Color(213, 236, 227));
+        this.player = new Player();
+        score = new GraphicsText();
+        lives = new GraphicsText();
+        level = new GraphicsText();
+        initialLives();
+        initialScore();
+        initialLevel();
+        setUpGame(); 
+    }
+
+    public void setUpGame() {
+        gameLayer = new GraphicsGroup();
+
+        mushroomManager = new MushroomManager(gameLayer);
+
+        snake = new Snake(Color.red, gameLayer);
+
+        canvas.add(gameLayer);
+
+        animating = true;
+
+        snake.getHead().moveAroundHelper(canvas);
+        
+        canvas.animate(() -> {if(animating) {
+            moveSnake();
+        }});
+    }
+    /**
+     * Manages snake movement and lives
+     */
+    public void moveSnake() {
+        if(!snake.bodyCollision() &&
+        !snake.wallCollision(CANVAS_WIDTH, CANVAS_HEIGHT) && player.showLives() > 0) {
+            if (testWin()) {
+                return;
+            }
+            snake.moveHead(canvas);
+            if (mushroomManager.findMushroomAtPosition(snake.getHead().getCenter())) {
+                player.addPoint();
+                updateScore();
+                snake.makeLonger(15);
+            }
+            snake.moveBody();
+        }
+        else if (snake.wallCollision(CANVAS_WIDTH, CANVAS_HEIGHT) || snake.bodyCollision()) {
+            snake.resetSnakeOnDeath(canvas);
+            player.loseLife();
+            updateLives();
+            return;
+        }
```

Comment:
It might be possible to tighten this up a bit — perhaps by having body segments, walls, and mushrooms all implement a `Collidable` interface.